<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ChecksumOptions</class>
 <widget class="QWidget" name="ChecksumOptions">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>558</width>
    <height>489</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configure Checksum</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="4" column="1">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>107</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page_crc">
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="3" column="1">
        <widget class="QLineEdit" name="lineEditFinalXOR"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>CRC polynomial (hex):</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Configure data ranges for CRC</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QTableView" name="tableViewDataRanges">
            <attribute name="horizontalHeaderHighlightSections">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderCascadingSectionResizes">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderHighlightSections">
             <bool>false</bool>
            </attribute>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QToolButton" name="btnAddRange">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset theme="list-add">
                <normaloff>../../../</normaloff>../../../</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btnRemoveRange">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset theme="list-remove">
                <normaloff>../../../</normaloff>../../../</iconset>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="lineEditStartValue"/>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>CRC type:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Final XOR (hex):</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Start value (hex):</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="lineEditCRCPolynomial"/>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="comboBoxCRCFunction"/>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QLabel" name="label_crc_info">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Order=17&lt;/p&gt;&lt;p&gt;Length of checksum=16&lt;/p&gt;&lt;p&gt;start value length =16&lt;/p&gt;&lt;p&gt;final XOR length = 16&lt;/p&gt;&lt;p&gt;Polynomial = x&lt;span style=&quot; vertical-align:super;&quot;&gt;1&lt;/span&gt; + 4&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_wsp">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The Wireless Short Packet (WSP) standard uses three different checksums. URH can automatically detect the used checksum algorithm from the message. However, you can enforce the usage of a certain checksum if you need to.&lt;/p&gt;&lt;p&gt;With &lt;span style=&quot; font-weight:600;&quot;&gt;Automatic&lt;/span&gt; setting, checksums are chosen by these rules (as defined in &lt;a href=&quot;http://hes-standards.org/doc/SC25_WG1_N1493.pdf&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;WSP standard&lt;/span&gt;&lt;/a&gt;):&lt;/p&gt;&lt;p&gt;1) 4 Bit Checksum - For Switch Telegram (RORG=5 or 6 and STATUS = 0x20 or 0x30)&lt;/p&gt;&lt;p&gt;2) 8 Bit Checksum: STATUS bit 2&lt;sup&gt;7&lt;/sup&gt; = 0&lt;/p&gt;&lt;p&gt;3) 8 Bit CRC: STATUS bit 2&lt;sup&gt;7&lt;/sup&gt; = 1&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignJustify|Qt::AlignVCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButtonWSPAuto">
         <property name="text">
          <string>Automatic (recommended)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButtonWSPChecksum4">
         <property name="text">
          <string>Force Checksum4</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButtonWSPChecksum8">
         <property name="text">
          <string>Force Checksum8</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButtonWSPCRC8">
         <property name="text">
          <string>Force CRC8</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QComboBox" name="comboBoxCategory">
     <item>
      <property name="text">
       <string>CRC</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Wireless Short Packet Checksum</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>Checksum category:</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>comboBoxCategory</tabstop>
  <tabstop>comboBoxCRCFunction</tabstop>
  <tabstop>lineEditCRCPolynomial</tabstop>
  <tabstop>lineEditStartValue</tabstop>
  <tabstop>lineEditFinalXOR</tabstop>
  <tabstop>tableViewDataRanges</tabstop>
  <tabstop>btnAddRange</tabstop>
  <tabstop>radioButtonWSPAuto</tabstop>
  <tabstop>btnRemoveRange</tabstop>
  <tabstop>radioButtonWSPChecksum4</tabstop>
  <tabstop>radioButtonWSPChecksum8</tabstop>
  <tabstop>radioButtonWSPCRC8</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
